library/core/src/slice/mod.rs

```diff
@@ -561,15 +561,47 @@ impl<T> [T] {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn swap(&mut self, a: usize, b: usize) {
-        // Can't take two mutable loans from one vector, so instead use raw pointers.
-        let pa = ptr::addr_of_mut!(self[a]);
-        let pb = ptr::addr_of_mut!(self[b]);
-        // SAFETY: `pa` and `pb` have been created from safe mutable references and refer
-        // to elements in the slice and therefore are guaranteed to be valid and aligned.
-        // Note that accessing the elements behind `a` and `b` is checked and will
-        // panic when out of bounds.
+        assert!(a < self.len());
+        assert!(b < self.len());
```

Comment:
the panic message degraded.

previously the out-of-bound panic would say like

```
index out of bounds: the len is 3 but the index is 4
```

but now it will just say

```
assertion failed: a < self.len()
```

Reply:
And tests didn't catch that, need test )